Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/ToolBar/EditorToolBar.cpp

```diff
@@ -135,42 +159,71 @@ namespace AzToolsFramework
 
                         m_toolBar->addAction(action);
                     }
-                    break;
                 }
+                break;
             case ToolBarItemType::Separator:
                 {
                     m_toolBar->addSeparator();
-                    break;
                 }
+                break;
+            case ToolBarItemType::ActionAndSubMenu:
             case ToolBarItemType::Widget:
                 {
                     m_toolBar->addAction(elem.second.m_widgetAction);
-                    break;
                 }
+                break;
             default:
                 break;
             }
         }
     }
 
-    EditorToolBar::ToolBarItem::ToolBarItem(ToolBarItemType type, AZStd::string identifier)
+    EditorToolBar::ToolBarItem::ToolBarItem(QToolBar* toolBar, ToolBarItemType type, AZStd::string identifier, AZStd::string subMenuIdentifier)
         : m_type(type)
+        , m_identifier(AZStd::move(identifier))
+        , m_subMenuIdentifier(AZStd::move(subMenuIdentifier))
     {
-        if (type != ToolBarItemType::Separator)
+        switch (m_type)
         {
-            m_identifier = AZStd::move(identifier);
+            case ToolBarItemType::Widget:
+                {
+                    if (QWidget* widget = m_actionManagerInternalInterface->GenerateWidgetFromWidgetAction(m_identifier))
+                    {
+                        m_widgetAction = new QWidgetAction(nullptr);
+                        m_widgetAction->setDefaultWidget(widget);
+                    }
+                }
+                break;
+
+            case ToolBarItemType::ActionAndSubMenu:
+                {
+                    if (QAction* action = m_actionManagerInternalInterface->GetAction(m_identifier);
+                        QMenu* subMenu = m_menuManagerInternalInterface->GetMenu(m_subMenuIdentifier))
```

Comment:
comment: Chaining the if initializer here with the second intialization is maybe a little confusing 😝 I love if-initializers more than most but I can see how this might be a little hard to grok on first glance 🤔 It maybe is fine as-is but might be worth separating potentially, totally up to you just wanted to mention it

Reply:
Will split!